ext/symengine/symengine.c

```diff
@@ -44,6 +45,9 @@ void Init_symengine() {
     rb_define_method(c_basic, "subs", cbasic_subs, -1);
     rb_define_method(c_basic, "coerce", cbasic_coerce, 1);
 
+    //Sympify as a Module Level Function
+    rb_define_module_function(m_symengine, "sympify", cutils_sympify, 1);
```

Comment:
I agree that sympify is not the best name. For the Python wrappers I think that's fine, since there we want to be compatible with SymPy anyway. But for Ruby, Julia or Haskell wrappers, I think a better name is needed.

All the options that you listed seem fine with me.